projects/plugins/wpcomsh/wpcom-themes/themes.php

```diff
@@ -5,174 +5,6 @@
- *
- * @param mixed  $res    The result object.
- * @param string $action The action.
- * @param mixed  $args   The arguments.
- *
- * @return mixed|stdClass
- */
-function wpcomsh_block_themes_wpcom_themes_api_result( $res, string $action, $args ) {
-	// Pre-requisites checks.
-	$tag = $args->tag ?? '';
-	if ( 'query_themes' !== $action || 'full-site-editing' !== $tag ) {
-		return $res;
-	}
-
-	$wpcom_themes_service = wpcomsh_get_wpcom_themes_service_instance();
-
-	// Add results to the resulting array.
-	return $wpcom_themes_service->filter_themes_api_result_block_themes( $res );
-}
-add_filter( 'themes_api_result', 'wpcomsh_block_themes_wpcom_themes_api_result', 0, 3 );
-
-/**
- * Process the themes API result and add WPCom themes when using the search.
- *
- * @param mixed  $res    The result object.
- * @param string $action The action.
- * @param mixed  $args   The arguments.
- *
- * @return mixed|stdClass
- */
-function wpcomsh_search_wpcom_themes_api_result( $res, string $action, $args ) {
-	// Pre-requisites checks.
-	$search = $args->search ?? '';
-	if ( 'query_themes' !== $action || '' === $search ) {
-		return $res;
-	}
-
-	$wpcom_themes_service = wpcomsh_get_wpcom_themes_service_instance();
-
-	// Add results to the resulting array.
-	return $wpcom_themes_service->filter_themes_api_result_search( $res, $search );
-}
-add_filter( 'themes_api_result', 'wpcomsh_search_wpcom_themes_api_result', 0, 3 );
-
-/**
- * Process the themes API result and add WPCom themes when using the filter feature.
- *
- * @param mixed  $res    The result object.
- * @param string $action The action.
- * @param mixed  $args   The arguments.
- *
- * @return mixed|stdClass
- */
-function wpcomsh_feature_filter_wpcom_themes_api_result( $res, string $action, $args ) {
-	// Pre-requisites checks.
-	$tags = $args->tag ?? array();
-	if ( 'query_themes' !== $action || ! $tags ) {
-		return $res;
-	}
-
-	if ( ! is_array( $tags ) ) {
-		$tags = array( $tags );
-	}
-
-	$wpcom_themes_service = wpcomsh_get_wpcom_themes_service_instance();
-
-	// Add results to the resulting array.
-	return $wpcom_themes_service->filter_themes_api_result_feature_filter( $res, $tags );
-}
-add_filter( 'themes_api_result', 'wpcomsh_feature_filter_wpcom_themes_api_result', 0, 3 );
-
-/**
- * Process the themes API result theme_information for WPCom themes if needed.
- *
- * @param mixed  $res    The result object.
- * @param string $action The action.
- * @param mixed  $args   The arguments.
- *
- * @return mixed|stdClass|WP_Error|null
- */
-function wpcomsh_theme_information_wpcom_themes_api_result( $res, string $action, $args ) {
-	// Pre-requisites checks.
-	if ( 'theme_information' !== $action ) {
-		return $res;
-	}
-
-	$wpcom_themes_service = wpcomsh_get_wpcom_themes_service_instance();
-
-	return $wpcom_themes_service->get_theme( $args->slug ) ?? $res;
-}
-add_filter( 'themes_api_result', 'wpcomsh_theme_information_wpcom_themes_api_result', 0, 3 );
-
```

Comment:
Should we move over this whole file? Or is what's left in this file separate enough that it's fine to leave in a separate location? As in, if someone needed to fix something related to the themes API, then they wouldn't need to work in both this file and `jetpack-mu-wpcom/src/features/wpcom-themes/wpcomsh-themes.php`?

Reply:
I reviewed more thoroughly. I think this file can stay here and the separation is good. What's left in this file isn't about API calls. It's about symlinking wpcom themes for atomic sites, and doesn't run on api calls.